src/hotspot/share/opto/regmask.hpp

```diff
@@ -68,24 +69,167 @@ class RegMask {
   static const unsigned int _RM_MAX      = _RM_SIZE - 1U;
 
   union {
-    // Array of Register Mask bits.  This array is large enough to cover
-    // all the machine registers and all parameters that need to be passed
-    // on the stack (stack registers) up to some interesting limit.  Methods
-    // that need more parameters will NOT be compiled.  On Intel, the limit
-    // is something like 90+ parameters.
+    // Array of Register Mask bits.  This array is large enough to cover all
+    // the machine registers and all parameters that need to be passed on the
+    // stack (stack registers) up to some interesting limit. On Intel, the
+    // limit is something like 90+ parameters.
     int       _RM_I[RM_SIZE];
     uintptr_t _RM_UP[_RM_SIZE];
   };
 
-  // The low and high water marks represents the lowest and highest word
+  // In rare situations (e.g., "more than 90+ parameters on Intel"), we need to
+  // extend the register mask with dynamically allocated memory.
+  uintptr_t* _RM_UP_EXT = nullptr;
```

Comment:
Have you considered using a growable array (`src/hotspot/share/utilities/growableArray.hpp`) for this part?

Reply:
No, I'll have a look and see if it makes sense to use growable arrays in this case. Thanks!

Reply:
Yes, it is good suggestion. Please look.

Reply:
I have investigated using a `GrowableArray` now, including a discussion with people involved in their development. Currently, there are some limitations with `GrowableArray` that have a negative performance impact for our use case:
- There is no efficient copy/clone operation.
- `GrowableArray` construction currently default-initializes everything within their capacity, which is unnecessary in our case (and probably unnecessary in general as well).

Yes, one option is to implement the above for `GrowableArray`. There is, however, already an ongoing discussion on how to best do this, and there are some pitfalls. I prefer to not start a more general and largely orthogonal discussion on `GrowableArray` changes here and suggest we update my current manual allocation to use a `GrowableArray` in a separate RFE, after `GrowableArray` has the required functionality.